Content.Server/MagicMirror/MagicMirrorSystem.cs

```diff
@@ -220,15 +243,19 @@ private void OnAddSlotDoAfter(EntityUid uid, MagicMirrorComponent component, Add
                 return;
         }
 
-        var marking = _markings.MarkingsByCategoryAndSpecies(category, component.Target.Value.Comp.Species).Keys.FirstOrDefault();
+        var target = component.Target.Value;
+        var humanoid = Comp<HumanoidAppearanceComponent>(target);
+        var marking = _markings.MarkingsByCategoryAndSpecies(category, humanoid.Species).Keys.FirstOrDefault();
+
         if (string.IsNullOrEmpty(marking))
         {
             return;
         }
 
-        _humanoid.AddMarking(component.Target.Value.Owner, marking, Color.Black);
+        _humanoid.AddMarking(component.Target.Value, marking, Color.Black);
+
+        UpdateInterface(uid, component.Target.Value, args.Message.Session);
 
```

Comment:
Are args.Target and component.Target supposed to be different? Not sure why it's being tracked twice.